frontend/src/scenes/system_status/systemStatusLogic.ts

```diff
@@ -0,0 +1,46 @@
+import api from 'lib/api'
+import { kea } from 'kea'
+
+interface Error {
+    detail: string
+    code: string
+}
+
+interface SystemStatus {
+    metric: string
+    value: string
+}
+
+export const systemStatusLogic = kea({
+    actions: {
+        setError: (error: Error) => ({ error }),
+        addSystemStatus: (systemStatus: SystemStatus) => ({ systemStatus }),
+    },
+    loaders: {
+        systemStatus: [
+            [],
+            {
+                loadSystemStatus: async () => {
+                    return (await api.get('_system_status')).results
+                },
+            },
+        ],
+    },
+    reducers: {
+        systemStatus: {
+            addSystemStatus: (state: Array<SystemStatus>, { systemStatus }) => [systemStatus, ...state],
+        },
+        error: [
+            false,
+            {
+                setError: (_, { error }) => error,
+            },
+        ],
+    },
```

Comment:
Hey! Great work with the logic! 

We are also using `kea-typegen` ([docs](https://kea.js.org/docs/guide/typescript)), which currently requires manually converting `kea({})` to `kea<systemStatusLogicType<SystemStatus>>({})`. This `systemStatusLogicType` should be automatically generated in the frontend/types folder and can be imported from there.

When that's in place, you can change the `systemStatus` loader's default value from `[]` to `[] as SystemStatus[]` and remove the manually defined type from `state` in the reducer.